presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -388,15 +390,17 @@ private TypeCompatibility typeCompatibilityForCovariantParametrizedType(Type fir
             return TypeCompatibility.incompatible();
         }
 
+        boolean firstCanCoerceToSecond = true;
```

Comment:
This seems unrelated to row type. Did you mean to put it in a previous commit?

Reply:
It is related.

The previous commit is just a refactor. It didn't introduce any semantic change. The assumption that `canCoerce` is implicitly defined by `getCommonSuperTypes` remains.

This commit introduced code that breaks the assumption. And this section of code is to deal with the broken assumption.

I could see the argument for moving this to the previous commit, but I try to keep the previous commit a refactor only.

Reply:
Ok, maybe change the commit title/description then? It mentions row type only, but this affects parametric types.